File: PropertEase.Core/Entities/ReservationNotification.cs
using PropertEase.Core.Entities.Base;
using PropertEase.Core.Entities.Identity;

namespace PropertEase.Core.Entities
{
    public class ReservationNotification : BaseEntity
    {
        public int UserId { get; set; }
        public ApplicationUser? User { get; set; }
        public int? ReservationId { get; set; }
        public PropertyReservation? Reservation { get; set; }
        public string Message { get; set; } = string.Empty;
        public bool IsSeen { get; set; } = false;
        public string? ReservationNumber { get; set; }
        public string? PropertyName { get; set; }
        public string? PropertyPhotoUrl { get; set; }
    }
}

File: UI/propertease_admin/lib/models/reservation_notification.dart
class ReservationNotification {
  final int? id;
  final int? userId;
  final int? reservationId;
  final String? title;
  final String? message;
  final bool? isSeen;
  final String? reservationNumber;
  final String? propertyName;
  final String? propertyPhotoUrl;
  final DateTime? createdAt;

  const ReservationNotification({
    this.id,
    this.userId,
    this.reservationId,
    this.title,
    this.message,
    this.isSeen,
    this.reservationNumber,
    this.propertyName,
    this.propertyPhotoUrl,
    this.createdAt,
  });

  factory ReservationNotification.fromJson(Map<String, dynamic> json) =>
      ReservationNotification(
        id:               json['id'] as int?,
        userId:           json['userId'] as int?,
        reservationId:    json['reservationId'] as int?,
        title:            json['title'] as String?,
        message:          json['message'] as String?,
        isSeen:           json['isSeen'] as bool?,
        reservationNumber: json['reservationNumber'] as String?,
        propertyName:     json['propertyName'] as String?,
        propertyPhotoUrl: json['propertyPhotoUrl'] as String?,
        createdAt: json['createdAt'] != null
            ? DateTime.tryParse(json['createdAt'] as String)
            : null,
      );
}
